src/js/core/MesosStream.ts

```diff
@@ -1,11 +1,90 @@
+import { Observable, of, timer } from "rxjs";
+import { fromFetch } from "rxjs/fetch";
+import { map, repeatWhen, shareReplay, switchMap } from "rxjs/operators";
 // @ts-ignore
-import { stream } from "@dcos/mesos-client";
-import { timer } from "rxjs";
-import { repeatWhen } from "rxjs/operators";
+import proto from "../mesos-proto";
+
+const eventCoder = proto.lookup("Event");
+
+const utf8Coder = new TextDecoder("utf-8");
+const recordIOChunk$ = (
+  reader: ReadableStreamDefaultReader<Uint8Array>
+): Observable<Uint8Array> => {
+  return new Observable(observer => {
+    // a buffer to temporarily hold binary data that is streamed. it might be
+    // that mesos streams less than a full chunk - that's why we need to store
+    // that temporarily.
+    let buffer = new Uint8Array();
+
+    // @ts-ignore
+    reader.read().then(function process({ done, value }) {
+      if (done) {
+        return observer.complete();
+      }
+
+      // append the data that's coming in to our buffer
+      const newBuffer = new Uint8Array(buffer.length + value.length);
+      newBuffer.set(buffer);
+      newBuffer.set(value, buffer.length);
+      buffer = newBuffer;
+
+      // check if we have enough data in the buffer to emit a chunk
+      const newlineIndex = buffer.indexOf(10) + 1;
+      const chunkLengthStr = utf8Coder.decode(buffer.slice(0, newlineIndex));
+      const chunkLength = parseInt(chunkLengthStr, 10);
+      if (buffer.length >= chunkLength) {
+        // emit a chunk
+        const chunk = buffer.slice(newlineIndex, chunkLength + newlineIndex);
+        observer.next(chunk);
+
+        // remove emitted chunk from buffer
+        buffer = buffer.slice(chunkLength + newlineIndex, buffer.length);
+      }
+
+      // handle next chunk
+      return reader.read().then(process);
+    });
+  });
+};
+
+// The `toObject` is needed to convert enums into readable strings
+const protobufToEvent = (chunk: Uint8Array) =>
+  eventCoder.toObject(eventCoder.decode(chunk), { enums: String });
+
+const mesos$ = fromFetch("/mesos/api/v1?subscribe", {
+  method: "POST",
+  body: JSON.stringify({ type: "SUBSCRIBE" }),
+  headers: {
+    Accept: "application/x-protobuf",
+    "Content-Type": "application/json"
+  }
+}).pipe(
+  switchMap(response => {
+    if (!response) {
+      throw Error("No response object returned for MesosStateStream");
+    }
+
+    // we only ever expect JSON to be served during integration-tests where
```

Comment:
Cypress is not as bad as it sounds in other projects we use it very effectively maybe we want to get some learnings from there and solve the problems we have in the cypress setup in dcos-ui. And let's please not forget that moving to another testing system also means that we need to rewrite a lot of tests. Let's keep on discussing this offline since it is not part of this PR.

Reply:
Well, it neither supports dynamic stubbing, modern browser APIs, nor multi browser testing and if we need to roll all of this our own than I don't see the value it adds. That being said I want us to consider dropping us that doesn't mean that we're going to as their may things it offers that are reason enough to keep it around.

Reply:
i think i like the idea. i'll have a look at what Julian once put together to get a feeling for what that means in daily development.